### PR TITLE
fix: reduce signcolumn/foldcolumn from window width

### PR DESCRIPTION
**Pull request by ssiyad**
https://github.com/nvim-tree/nvim-tree.lua/issues/1679#issuecomment-1291030546

### CODE REVIEW

**Comment by pleshevskiy** [comment]
I've checked and it works! Thanks!

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Looking good:

20221029_135419 441x477

Please be more defensive:

- [ ] `win_getid` may return 0 which will result in an incorrect result from `getwininfo`
- [ ] `getwininfo` may return an empty list

`textoff` is OK as it is "guaranteed" to be a number for a successful result.

File: lua/nvim-tree/renderer/components/full-name.lua [hunk 1]
```diff
@@ -32,14 +54,17 @@ local function show()
     return
   end
 
-  local width = fn.strdisplaywidth(fn.substitute(line, "[^[:print:]]*$", "", "g"))
-  if width < fn.winwidth(0) then
+  local text_width = fn.strdisplaywidth(fn.substitute(line, "[^[:print:]]*$", "", "g"))
+  local win_width = effective_win_width()
```
Comment (alex-courtis) [inline lua/nvim-tree/renderer/components/full-name.lua:58]: nice

**Review by alex-courtis — APPROVED** [review]
Works beautifully, thank you for the contribution.

**Pull request merged**